pkg/admissioncontroller/webhook/admission/auditpolicy/handler.go

```diff
@@ -252,13 +251,15 @@ func validateAuditPolicySemanticsForKubernetesVersion(auditPolicy string, kubern
 		return http.StatusUnprocessableEntity, fmt.Errorf("provided invalid audit policy: %v", errList)
 	}
 
-	if kubernetesVersion != nil && schemaVersion.Version != "v1" {
-		v, err := semver.NewVersion(*kubernetesVersion)
-		if err != nil {
-			return http.StatusUnprocessableEntity, fmt.Errorf("unable to build version: %w", err)
-		}
-		if !version.ConstraintK8sLess124.Check(v) {
-			return http.StatusUnprocessableEntity, fmt.Errorf("audit policy with apiVersion '%s' is not supported for kubernetes version >= 1.24.0", schemaVersion.Version)
+	if len(shootToKubernetesVersionMap) > 0 && schemaVersion.Version != "v1" {
```

Comment:
```suggestion
	if schemaVersion.Version != "v1" {
```

If the map is empty then below loop will do nothing anyways.